src/SQLite.cs

```diff
@@ -1946,77 +1941,440 @@ protected internal void Dispose()
-                if (!Indices.Any()
-                    && !IsPK
-                    && ((createFlags & CreateFlags.ImplicitIndex) == CreateFlags.ImplicitIndex)
-                    && Name.EndsWith (Orm.ImplicitIndexSuffix, StringComparison.OrdinalIgnoreCase)
-                    )
-                {
-                    Indices = new IndexedAttribute[] { new IndexedAttribute() };
-                }
-                IsNullable = !(IsPK || Orm.IsMarkedNotNull(prop));
-                MaxStringLength = Orm.MaxStringLength(prop);
+		public bool IsAutoInc { get; set; }
+		public bool IsAutoGuid { get; set; }
 
-                StoreAsText = prop.PropertyType.GetTypeInfo().GetCustomAttribute(typeof(StoreAsTextAttribute), false) != null;
-            }
+		public bool IsPK { get; set; }
+
+		public IEnumerable<IColumnIndex> Indices { get; set; }
+
+		public bool IsNullable { get; set; }
 
-			public void SetValue (object obj, object val)
+		public int? MaxStringLength { get; set; }
+
+		public bool StoreAsText { get; set; }
+
+		public ColumnMapping(PropertyInfo prop)
+		{
+			_prop = prop;
+		}
+
+		public void SetValue(object obj, object val)
+		{
+			_prop.SetValue(obj, val, null);
+		}
+
+		public object GetValue(object obj)
+		{
+			return _prop.GetValue(obj, null);
+		}
+	}
+
+	public class ColumnMappingFromAttributes : ColumnMapping
+	{
+		public ColumnMappingFromAttributes(PropertyInfo prop, CreateFlags createFlags = CreateFlags.None) : base(prop)
+		{
+			var colAttr = (ColumnAttribute)prop.GetCustomAttributes(typeof(ColumnAttribute), true).FirstOrDefault();
+
+			Name = colAttr == null ? prop.Name : colAttr.Name;
+			//If this type is Nullable<T> then Nullable.GetUnderlyingType returns the T, otherwise it returns null, so get the actual type instead
+			ColumnType = Nullable.GetUnderlyingType(prop.PropertyType) ?? prop.PropertyType;
+			Collation = Orm.Collation(prop);
+
+			IsPK = Orm.IsPK(prop) ||
+				(((createFlags & CreateFlags.ImplicitPK) == CreateFlags.ImplicitPK) &&
+					 string.Compare(prop.Name, Orm.ImplicitPkName, StringComparison.OrdinalIgnoreCase) == 0);
+
+			var isAuto = Orm.IsAutoInc(prop) || (IsPK && ((createFlags & CreateFlags.AutoIncPK) == CreateFlags.AutoIncPK));
+			IsAutoGuid = isAuto && ColumnType == typeof(Guid);
+			IsAutoInc = isAuto && !IsAutoGuid;
+
+			Indices = Orm.GetIndices(prop);
+			if (!Indices.Any()
+				&& !IsPK
+				&& ((createFlags & CreateFlags.ImplicitIndex) == CreateFlags.ImplicitIndex)
+				&& Name.EndsWith(Orm.ImplicitIndexSuffix, StringComparison.OrdinalIgnoreCase)
+				)
 			{
-				_prop.SetValue (obj, val, null);
+				Indices = new IndexedAttribute[] { new IndexedAttribute() };
+			}
+			IsNullable = !(IsPK || Orm.IsMarkedNotNull(prop));
+			MaxStringLength = Orm.MaxStringLength(prop);
+
+			StoreAsText = prop.PropertyType.GetTypeInfo().GetCustomAttribute(typeof(StoreAsTextAttribute), false) != null;
+		}
+	}
+
+	public static class TableMappingFluentExtensions
```

Comment:
These should not be public. I don't like introducing APIs unrelated to this library.